modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py

```diff
@@ -1976,27 +2066,71 @@ def _execute_arrow(self):
         pyarrow.Table
             The resulting table.
         """
-        if isinstance(self._op, FrameNode):
-            if self._partitions.size == 0:
-                return pyarrow.Table.from_pandas(pd.DataFrame({}))
+        result = None
+        stack = [self]
+
+        while stack:
```

Comment:
I'm kind of afraid to merge the changes that completely rewrite essential logic one day before the release. I'm not sure that our test suites are capable to test all variations of computation trees (as usually we only run 1-2 operations for the test, so the tree can't grow any bigger than that).

As far as I understand everything used to work even without getting rid of the recursions here, so can we detach these changes into a separate PR and move it to a next release where we would have more time to test it properly? cc @Garra1980

Reply:
The logic should not be changed. The tree should be traversed in the same order, but non-recursively.
I agree, that it should not be merged on the eve of the forthcoming release.